CI Linux usage is broken

<!-- Welcome! Thank you for contributing. These HTML comments will not render in the issue.

Before creating a new issue:
* Search for relevant issues
* Follow the issue reporting guidelines:
https://jupyterlab.readthedocs.io/en/latest/getting_started/issue.html
-->

## Description

<!--Describe the bug clearly and concisely. Include screenshots if possible-->
The following error is emitted:

```
Error: Cannot find module '/home/runner/work/jupyterlab/jupyterlab/jupyterlab/examples/federated/core_package/package.json'
```

## Root cause

The error comes from 

https://github.com/jupyterlab/jupyterlab/blob/74405d37ad156d8bdc0e7c36f199abc1eb642c1f/jupyterlab/federated_labextensions.py#L215

The `ext_path` is not correct to resolve relative path `core_path` as it is defined relative to the command execution directory not the extension directory.

I have no clue why it works before.

--- a/jupyterlab/federated_labextensions.py
+++ b/jupyterlab/federated_labextensions.py
@@ -14,11 +14,13 @@ import platform
 import shutil
 import subprocess
 import sys
+from pathlib import Path
 
 try:
     from importlib.metadata import PackageNotFoundError, version
 except ImportError:
     from importlib_metadata import PackageNotFoundError, version
+
 from os.path import basename
 from os.path import join as pjoin
 from os.path import normpath
@@ -197,7 +199,10 @@ def build_labextension(
     """Build a labextension in the given path"""
     if core_path is None:
         core_path = osp.join(HERE, "staging")
-    ext_path = osp.abspath(path)
+    else:
+        core_path = str(Path(core_path).resolve())
+
+    ext_path = str(Path(path).resolve())
 
     if logger:
         logger.info("Building extension in %s" % path)
@@ -221,7 +226,9 @@ def watch_labextension(
     """Watch a labextension in a given path"""
     if core_path is None:
         core_path = osp.join(HERE, "staging")
-    ext_path = osp.abspath(path)
+    else:
+        core_path = str(Path(core_path).resolve())
+    ext_path = str(Path(path).resolve())
 
     if logger:
         logger.info("Building extension in %s" % path)

